main.rb

```diff
@@ -0,0 +1,123 @@
+# frozen_string_literal: true
+
+# rubocop:disable Metrics/PerceivedComplexity,Metrics/CyclomaticComplexity
+# rubocop:disable Metrics/MethodLength:
+# rubocop:disable Metrics/AbcSize
+
+# This module is a creation of the original Enumerable
+module Enumerable
+  def my_each
+    return to_enum :my_each unless block_given?
+
+    a = self
+    a = *self if is_a? Range
+    a.length.times { |i| yield(a[i]) }
+    a
+  end
+
+  def my_each_with_index
+    return to_enum :my_each_with_index unless block_given?
+
+    a = self
+    a = *self if is_a? Range
+    a.length.times { |i| yield(i, a[i]) }
+    a
+  end
```

Comment:
- For my_each_with_index, it should call the given block once for each element in self.
According to the enumerable guide [here](https://ruby-doc.org/core-2.6.5/Enumerable.html#method-i-each_with_index), the item comes before the index, however, in your method, (line 23), the index comes before the item.

For example
`[1,2,3,4].my_each_with_index{ |num, idx| p "Num: #{num}, idx: #{idx}\n" }`
The built in method returns
`"Num: 0, idx: 1\n"
"Num: 1, idx: 2\n"
"Num: 2, idx: 3\n"
"Num: 3, idx: 4\n"
[1, 2, 3, 4]`

`[1,2,3,4].my_each_with_index{ |num, idx| p "Num: #{num}, idx: #{idx}\n" }`
Yours returns
`"Num: 1, idx: 0\n"
"Num: 2, idx: 1\n"
"Num: 3, idx: 2\n"
"Num: 4, idx: 3\n"
[1, 2, 3, 4]
`